sdk/dotnet/Pulumi/Deployment/Outputs.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016-2022, Pulumi Corporation
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// A type-safe wrapper around <see cref="IDictionary{String, Object}"/> for generating stack outputs from created resources. 
+    /// </summary>
+    public class Outputs : IEnumerable<KeyValuePair<string, object?>>
+    {
+        // Internally, we keep track of a dictionary which keeps track of untyped values (objects)
+        private readonly Dictionary<string, object?> outputs = new Dictionary<string, object?>();
+
+        public void Add(string outputName, string value)
```

Comment:
@Frassle that's the thing, I am not sure on this one from the users perspective 🤔 if you add `Output<T>` overload, users are able to do this:
```cs
return new Outputs 
{
   { "Bucket", new Bucket("my-bucket") } 
};
```
And yes, users _can_ serialize resources as I've seen in the code but do we want to expose this capability when outputting stack values? I went through the examples to see if there is anything using this pattern but couldn't find anything where users need the full scope of outputs

Reply:
Yeh it's an odd space. If we had extensible traits we could mark this as any pulumi serializable value, but given the type system isn't that strong I don't think we can do any better than just `object` which renders this just a slightly nicer IDictionary<string,object>?

Reply:
> I don't think we can do any better than just object which renders this just a slightly nicer IDictionary<string,object>

Yeah I agree. I think using a couple of types for now will be more than enough to cover tons of use cases without going with "outputs can be anything". I am still in favor of adding `Outputs` and actually pushing its use with a lambda in `RunAsync` when migrating examples/docs to .NET 6 because right now dotnet samples _stand out_ with the extra boilerplate when compared to the other languages